probe-rs/src/architecture/riscv/communication_interface.rs

```diff
@@ -100,6 +102,23 @@ enum DebugModuleVersion {
     NonConforming = 15,
 }
 
+#[repr(u8)]
+#[derive(Debug, Copy, Clone)]
+enum CoreRegisterAbstractCmdSupport {
+    Read = 1 << 0,
+    Write = 1 << 1,
+    Both = Self::Read as u8 | Self::Write as u8,
+}
+
+impl CoreRegisterAbstractCmdSupport {
+    fn supports(&self, o: Self) -> bool {
+        o as u8 & *self as u8 == o as u8
+    }
+    fn unset(&mut self, o: Self) {
+        *self = unsafe { mem::transmute(*self as u8 & !(o as u8)) }
+    }
+}
```

Comment:
Please implement this without using unsafe. We'd like to keep probe-rs free of unsafe, as much as possible.

The current implementation can cause undefined behaviour, if you try to set both `Read` and `Write` to unsupported. 
The resulting value for the enum will be 0, which is not a valid value. See this [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=868fb6a3eef420ee46126091e5ffe7fe), if you run the code the output will be wrong, and if you run it with Miri (in the Tools menu), it will tell you that the program caused undefined behaviour.

Reply:
I will use a bitfield instead.
By the way, good catch, I feel really stupid with the 0 value. Should not use unsafe code for stuff like this.